cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java

```diff
@@ -205,7 +208,9 @@ public boolean createFeatureEntity(String featureEntityName, boolean unique, int
     final AttributeDefinition[] attributes = Arrays.stream(featureNames)
         .map(s -> new AttributeDefinition(s, VECTOR, length))
         .toArray(AttributeDefinition[]::new);
-    return this.createFeatureEntity(featureEntityName, unique, attributes);
+    var feature = this.createFeatureEntity(featureEntityName, unique, attributes);
+    this.createIndex(featureEntityName, GENERIC_ID_COLUMN_QUALIFIER, unique ? IndexType.HASH_UQ : IndexType.HASH, null);
```

Comment:
It might be worth having indexes for the other fields as well, at least for the object id.

Reply:
Feature entities as create in this method only have the segment id. the object id is indexed in the segments entity https://github.com/vitrivr/cineast/blob/a2ced2934afdc947d56f471f59bf220dce0c09ef/cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java#L128

If you mean that we should index the field partially, i'm not sure if Cottontail supports that.

Reply:
I was specifically referring to the segments entity, the comment might have ended up on the wrong line. It might be worth indexing all the columns of the 'book-keeping' entities since they might be used for different lookups.

Reply:
Yes, that's already being done, see https://github.com/vitrivr/cineast/blob/a2ced2934afdc947d56f471f59bf220dce0c09ef/cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java#L145